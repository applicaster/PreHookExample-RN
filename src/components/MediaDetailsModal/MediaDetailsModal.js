import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Modal,
} from 'react-native';
import MediaDetailsScreen from '../MediaDetailsScreen';

class MediaDetailsModal extends Component {
  constructor(props) {
    super(props);
    this.closeModal = this.closeModal.bind(this);
  }

  closeModal() {
    const { toggleModal } = this.props;
    toggleModal({ modal: 'MediaModal' });
  }

  render() {
    const { isMediaModalVisible } = this.props;
    return (
      <Modal
        animationType={'fade'}
        transparent={false}
        visible={isMediaModalVisible}
        onRequestClose={this.closeModal}
      >
        <MediaDetailsScreen />
      </Modal>
    );
  }
}

MediaDetailsModal.propTypes = {
  isMediaModalVisible: PropTypes.bool,
  toggleModal: PropTypes.func,
};

export default MediaDetailsModal;
