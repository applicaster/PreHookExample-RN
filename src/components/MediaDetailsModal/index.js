import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { toggleModal } from '../../actions';
import {
  getMediaModalVisibility,
} from '../../selectors';

import MediaDetailsModal from './MediaDetailsModal';

const mapStateToProps = state => ({
  isMediaModalVisible: getMediaModalVisibility(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleModal,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(MediaDetailsModal);
