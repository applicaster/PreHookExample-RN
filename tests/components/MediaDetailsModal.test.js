import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import MediaDetailsModal from '../../src/components/MediaDetailsModal/MediaDetailsModal';

jest.mock('@applicaster/react-native-transformable-image', () => 'TransformableImage');
jest.mock('../../src/components/MediaDetailsScreen', () => 'MediaDetailsScreen');

describe('MediaDetailsModal', () => {
  test('component renders correctly when it is visible', () => {
    const html = renderer.create(
      <MediaDetailsModal isMediaModalVisible />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });

  test('component renders correctly when it is invisible', () => {
    const html = renderer.create(
      <MediaDetailsModal isMediaModalVisible={false} />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
