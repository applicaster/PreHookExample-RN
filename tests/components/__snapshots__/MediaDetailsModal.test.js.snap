// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaDetailsModal component renders correctly when it is invisible 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={false}
>
  <MediaDetailsScreen />
</Modal>
`;

exports[`MediaDetailsModal component renders correctly when it is visible 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={false}
  visible={true}
>
  <MediaDetailsScreen />
</Modal>
`;
